app/utils/modify-shared-ways.js

```diff
@@ -32,21 +35,29 @@ export default function modifySharedWays (sharedWays, node, coordinates, action)
         if (newWay.geometry.type === 'Polygon') {
           newWay.geometry.coordinates[0].splice(indexOfNodeInWay, 1)
         }
+
+        index = indexOfNodeInWay
         modifiedSharedWays.push(newWay)
       })
       break
 
     case 'ADD':
       if (node.properties.edge) {
         const indexOfPointOnEdge = node.properties.index
+
+        // QUESTION
+        // it seems like pointOnEdgeAtIndex is always the same
+        // i'm not sure if that's because indexOfPointOnEdge is supposed to be the same each time?
```

Comment:
I'm not sure what the deal is with this. There's a chance that this logic doesn't have the purpose that I think it does.

Reply:
You're right that the index here is always 0, that's because it's the index on the edge with 2 points. I think that's correct.